app/src/main/java/com/example/bigowlapp/activity/ScheduleReportActivity.java

```diff
@@ -0,0 +1,102 @@
+    private String scheduleUid, supervisorId;
+    private TextView scheduleReportTitle, scheduleReportStartTime, scheduleReportEndTime, scheduleReportLocation;
+    private ScheduleReportViewModel scheduleReportViewModel;
+    private ListView scheduleReportMemberListView;
+    private ScheduleReportMembersAdapter scheduleReportMembersAdapter;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        scheduleUid = getIntent().getStringExtra("scheduleUid");
+        supervisorId = getIntent().getStringExtra("supervisorId");
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+
+        if (scheduleReportViewModel == null) {
+            scheduleReportViewModel = new ViewModelProvider(this).get(ScheduleReportViewModel.class);
+        }
+
+        subscribeToData();
+    }
+
+    private void subscribeToData() {
+        if (!scheduleReportViewModel.isCurrentUserSupervisor(supervisorId)) {
+            return;
+        }
+
+        scheduleReportViewModel.getCurrentScheduleData(scheduleUid).observe(this, schedule -> {
+            if (schedule != null) {
+                scheduleReportTitle = findViewById(R.id.schedule_report_title);
+                scheduleReportStartTime = findViewById(R.id.schedule_report_start_time);
+                scheduleReportEndTime = findViewById(R.id.schedule_report_end_time);
+                scheduleReportLocation = findViewById(R.id.schedule_report_location);
+                scheduleReportMemberListView = findViewById(R.id.schedule_report_member_list);
+
+                scheduleReportTitle.setText(schedule.getTitle());
+                scheduleReportStartTime.setText(schedule.getStartTime().toDate().toString());
+                scheduleReportEndTime.setText(schedule.getEndTime().toDate().toString());
+                scheduleReportLocation.setText(getScheduleLocation(schedule.getLocation()));
+
+                scheduleReportViewModel.getScheduleMemberNameMap(schedule.getMemberList()).observe(this, memberNameMap -> {
+                    if (schedule.scheduleCurrentState() == Schedule.Status.ON_GOING)
+                        Toast.makeText(this, "Schedule is ongoing and attendance results are subject to change", Toast.LENGTH_LONG).show();
+
+                    scheduleReportMembersAdapter = new ScheduleReportMembersAdapter(this, memberNameMap, schedule);
+                    scheduleReportMemberListView.setAdapter(scheduleReportMembersAdapter);
+                });
+            }
+        });
+    }
+
+    private String getScheduleLocation(GeoPoint geoPoint) {
+        try {
+            return new Geocoder(this).getFromLocation(
+                    geoPoint.getLatitude(),
+                    geoPoint.getLongitude(),
+                    1)
+                    .get(0).getAddressLine(0);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return "ERROR";
+    }
```

Comment:
I think this should be moved to another class named LocationFormatter.

This might be useful for an old feature: View Schedules for members. When a member received a schedule, the schedule doesn't contain the location details, and this may useful for improving that feature. Don't fix the feature, but move this to another class so that fixing that feature would make things easier.